sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/document_producer.py

```diff
@@ -58,7 +58,13 @@ def __init__(self, partition_key_target_range, client, collection_link, query, d
         collection_id = _base.GetResourceIdOrFullNameFromLink(collection_link)
 
         def fetch_fn(options):
-            return self._client.QueryFeed(path, collection_id, query, options, partition_key_target_range["id"])
+            try:
+                return self._client.QueryFeed(path, collection_id, query, options, partition_key_target_range["id"])
+            except exceptions.CosmosHttpResponseError as e:
+                if e.status_code == http_constants.StatusCodes.GONE:
```

Comment:
same question as above

Reply:
seems like this one is necessary, otherwise the error does not get handled properly and instead gets stuck within the code execution (not thrown but not used)

Reply:
i was wrong ignore me